packages/ui/DatePicker/index.tsx

```diff
@@ -51,18 +57,17 @@ export const CustomDatePicker = ({
           strategy: "fixed",
         }}
         onSelect={onSelect}
-        selected={date}
-        name={placeholder}
+        selected={selected}
+        name={label}
         onChange={(date: Date) => {
-          setDate(date);
           onSelect(date);
         }}
         autoComplete="off"
-        placeholderText={placeholder}
         showYearDropdown
         isClearable
         onFocus={(e) => (e.target.readOnly = true)}
         dropdownMode="select"
+        customInput={<Input label={label} hasError={hasError} />}
```

Comment:
changes to fix the clear icon in date picker inputs (4/6)

```suggestion
        customInput={<Input label={label} hasError={hasError} onInputCleared={() => onSelect(null)}/>}
```